pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/util/SlingPathSuffix.java

```diff
@@ -0,0 +1,88 @@
+package com.redhat.pantheon.servlet.util;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents the suffix of a sling url and allows to extract information from it.
+ * To be used primarily in servlets where suffix information contains input information
+ * (i.e. parameters).
+ * <br><br>
+ *
+ * Parameters may be extracted like so:
+ *
+ * <pre>{@code
+ * SlingPathSuffix suffix = new SlingPathSuffix("/path/with/{param1}/and/{param2}");
+ * suffix.getParam("param1", request);
+ * suffix.getParam("param2", request);
+ * }</pre>
+ *
+ * The suffix needs a template or pattern to be provided in order to be able to extract
+ * information from it. It may be a completely static pattern or contain placeholders for
+ * named parameters.
+ *
+ * @author Carlos Munoz
+ */
+public class SlingPathSuffix {
+
+    private static final Pattern PARAMETER_PATTERN = Pattern.compile("(\\{[a-zA-Z0-9]+})");
```

Comment:
do we allow '-' in the sufix pattern?

Reply:
**Yes** in the pattern (the url itself). **No** for the parameter names.